lib/vector-buffers/src/variants/disk_v2/writer.rs

```diff
@@ -467,6 +467,9 @@ where
         // the actual encoded size and then check it against the limit.
         //
         // C'est la vie.
+        // Clone record here because encode will consume it and ack it to source.
+        // We have not yet archived the record, so we should avoid acknowledging the event.
+        let old_record = record.clone();
```

Comment:
Will this cause a double ACK since we are cloning the object? I wonder if it's simpler to not let do anything in line 538, check for inconsistent decode/encoding earlier and let destructor be run when record is referenced again in line 541

Reply:
cause of issue, is encoding. as soon as  you encode you acked because encode take ownership and destructs. 

No making clone wont do double ACK. Notification receiver is share pointer so it does not copy and it aggregate all ack. when last ref of share pointer goes ack happens.

I think rather than copying whole event, I can just take ref of notifier and attach it back. that would be performant.
Let me try.

Reply:
hmm it seems it is not staright forward to get hold of notifier and clone only that. we have top clone the complete event.